src/Raven.Server/Documents/DocumentCompare.cs

```diff
@@ -11,8 +11,15 @@ namespace Raven.Server.Documents
 {
     public static class DocumentCompare
     {
+        public struct DocumentCompareOptions
+        {
+            public bool TryMergeMetadataConflicts;
+            public bool ThrowOnAttachmentModifications;
+            public string DocumentId;
```

Comment:
Am I missing something, or is this never set?

Reply:
I forgot to set it but if we will use this than we can't use constant options so its allocation vs better erroring, i think it is best to catch the exception on patch only and add a better message there

Reply:
Sounds good, yes